src/services/ActivityService.ts

```diff
@@ -110,4 +107,119 @@ export class ActivityService {
                 throw new HTTPResponse(error.statusCode, { error: `${error.code}: ${error.message} At: ${error.hostname} - ${error.region} Request id: ${error.requestId}` });
             });
     }
+
+    /**
+     * Updates an activity in the database.
+     * @param activity - the payload containing the activity
+     * @returns Promise - void
+     */
+    public async updateActivity(activities: IActivity[]): Promise<void> {
+        const activitiesList: any[] = [];
+        for (const each of activities) {
+            // Payload validation
+            const validation: Joi.ValidationResult<IActivity> = Joi.validate(each, ActivityUpdateSchema);
+            if (validation.error) {
+                const error: string = validation.error.details[0].message;
+
+                throw new HTTPResponse(400, {error});
+            }
+            await this.dbClient.get({id: each.id})
```

Comment:
Would personally refactor this to:
    const result = await this.cdbClient.get(....).catch([catch stuff])
    [everything inside the "then"]

but this pattern is experimental within this project, so perhaps not appropriate for right now

Reply:
@adpetrescu thoughts? I'm fine without it, and can prototype in test-results for 5312

Reply:
Looks like we would not require catch too. it works fine without that. We could just do:
const result = await this.cdbClient.get(....);
[everything inside the "then"]

Not aware about how await handles errors/exceptions being thrown, but works without catching them. Can read more on this for better clarity.